qemu/tests/block_discard_hotplug.py

```diff
@@ -0,0 +1,126 @@
+"""
+hot-plug discard disk testing
+"""
+import logging
+
+from avocado.utils import process
+from virttest.utils_misc import get_linux_drive_path
+
+from virttest import storage
+from virttest import error_context
+from provider.block_devices_plug import BlockDevicesPlug
+from virttest import data_dir
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu discard hotplug support test:
+
+    1) Boot vm
+    2) Hot-plug the data disk with discard option
+    3) Format the data disk and mount it in guest
+    4) Execute dd command to the mounted disk in guest
+    5) Check disk size allocation
+    6) Execute rm and fstrim command in guest
+    7) Check disk size allocation
+    8) Hot-unplug the data disk
+    9) Reboot vm
+
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_scsi_debug_disk(guest_session=None):
+        """"
+        Get scsi debug disk on host or on guest which created as scsi-block.
+        """
+        cmd = "lsblk -S -n -p|grep scsi_debug"
+
+        if guest_session:
+            status, output = guest_session.cmd_status_output(cmd)
+        else:
+            status, output = process.getstatusoutput(cmd)
+
+        if status != 0:
+            test.fail("Can not find scsi_debug disk")
+
+        return output.split()[0]
+
+    def get_guest_discard_disk():
+        """
+        Get discard disk on guest.
+        """
+        if scsi_debug == "yes":
+            return get_scsi_debug_disk(session)
+
+        return get_linux_drive_path(session, disk_serial)
+
+    def check_disk_allocation():
+        """
+        Get the disk size allocation
+        """
+        if scsi_debug == "yes":
+            cmd = "cat /sys/bus/pseudo/drivers/scsi_debug/map"
+            output = process.system_output(cmd).decode().split(",")
+            return sum([abs(eval(i)) for i in output if i != ""])
+
+        cmd = "stat -c %b " + disk_name
+        return int(process.system_output(cmd).decode())
+
+    vm_name = params["main_vm"]
+    scsi_debug = params.get("scsi_debug", "no")
+    data_tag = params["data_tag"]
+    disk_serial = params["disk_serial"]
+
+    vm = env.get_vm(vm_name)
+    vm.verify_alive()
+
+    if scsi_debug == "yes":
+        disk_name = get_scsi_debug_disk()
+        vm.params["image_name_%s" % data_tag] = disk_name
+    else:
+        image_params = params.object_params(data_tag)
+        disk_name = storage.get_image_filename(image_params,
+                                               data_dir.get_data_dir())
+
+    timeout = float(params.get("login_timeout", 240))
+    session = vm.wait_for_login(timeout=timeout)
+    plug = BlockDevicesPlug(vm)
+    error_context.context("Hot-plug discarded disk in guest.", logging.info)
+    plug.hotplug_devs_serial(data_tag)
+
+    guest_disk_name = get_guest_discard_disk()
```

Comment:
```suggestion
    plug.hotplug_devs_serial(data_tag)
    guest_disk_name = '/dev/' + plug[0]
```
if you call the `plug.hotplug_devs_serial`, you could get the hotplugged disk id by `plug[index]`.

Reply:
Will update